### PR TITLE
#350: CI docker image testing

### LINKED ISSUE #350: CI docker image testing
This issue serves to experiment with using a Docker image of base dependencies in the NimbleSM CI testing to be pulled from dockerhub.

### PR DESCRIPTION
**Pull request by pierrepebay**
Work in conjunction with https://github.com/NimbleSM/DockerTestImages/pull/2

### CODE REVIEW

File: ci/build.sh [hunk 1]
```diff
@@ -1,7 +1,36 @@
...
+elif [ "$NimbleSM_ENABLE_MPI" = "OFF" ] && [ "$NimbleSM_ENABLE_TRILINOS" = "ON" ] && [ "$NimbleSM_ENABLE_KOKKOS" = "OFF" ] && [ "$NimbleSM_ENABLE_ARBORX" = "OFF" ]; then \
+     echo "Using SERIAL+TRILINOS NimbleSM configuration."; \
+     export NimbleSM_SPACK_ENV_NAME="nimble-serial-trilinos"; \
+elif [ "$NimbleSM_ENABLE_MPI" = "OFF" ] && [ "$NimbleSM_ENABLE_TRILINOS" = "OFF" ] && [ "$NimbleSM_ENABLE_KOKKOS" = "OFF" ] && [ "$NimbleSM_ENABLE_ARBORX" = "OFF" ]; then \
+     echo "Using SERIAL NimbleSM configuration."; \
+     export NimbleSM_SPACK_ENV_NAME="nimble-serial"; \
+else \
+     echo "No existing NimbleSM configuration with provided arguments. Exiting."; \
+     exit 1; \
+fi
```
Comment (nmm0) [inline ci/build.sh]: Maybe we can use the scope name given us to an environment variable here: https://github.com/NimbleSM/NimbleSM/blob/develop/.github/workflows/ubuntu20.04-gcc11-x64.yml#L34

Or construct the name of the environment in a similar manner

**Review by nmm0 — CHANGES_REQUESTED** [review]
Excellent, just some minor comments

File: ci/build.sh (hunk 1 above)
Comment (pierrepebay) [reply ci/build.sh]: That's a great idea, and to make things as clear as possible, I can change the spack environment names to the possible `SCOPE_NAME`s that way we don't even need this big if statement.

Comment (pierrepebay) [reply ci/build.sh]: https://github.com/NimbleSM/DockerTestImages/actions/runs/5826468563/job/15800501725#step:5:2213
`Error: 'pre_NimbleSMSerial+Trilinos': names must start with a letter, and only contain letters, numbers, _, and -` so this little trick can't happen 😔. I'll just use the `SCOPE_NAME` in the if statement.

File: ci/ubuntu20.04-gcc11-x64.dockerfile [hunk 2]
```diff
@@ -1,100 +1,22 @@
-FROM ubuntu:20.04 as build_dependencies-stage
-
+FROM pierrpebay/nimblesm:pr-2 AS build_nimble-stage
```
Comment (nmm0) [inline ci/ubuntu20.04-gcc11-x64.dockerfile]:
```suggestion
FROM nmm0/nimblesm-base-env:main AS build_nimble-stage
```

**Review by nmm0 — APPROVED** [review]
Looks good, thank you Pierre :)

**Pull request merged**